src/test/rgw/bucket_notification/test_bn.py

```diff
@@ -2266,111 +2204,38 @@ def test_ps_s3_multipart_on_master_http():
 
     # check http receiver
     keys = list(bucket.list())
-    print('total number of objects: ' + str(len(keys)))
-    events = http_server.get_and_reset_events()
-    for event in events:
-        assert_equal(event['Records'][0]['opaqueData'], opaque_data)
-        assert_true(event['Records'][0]['s3']['object']['eTag'] != '')
+    receiver.verify_s3_events(keys, exact_match=True, deletions=False)
 
     # cleanup
+    s3_notification_conf.del_config()
+    topic_conf.del_config()
+    # delete objects
     for key in keys:
         key.delete()
-    [thr.join() for thr in client_threads]
-    topic_conf.del_config()
-    s3_notification_conf.del_config(notification=notification_name)
     # delete the bucket
     conn.delete_bucket(bucket_name)
-    http_server.close()
-
+    receiver.close(task)
```

Comment:
you shoudl also check if `task != None` and call `task.stop()`

Reply:
there is not stop function for task